src/app/technical-record/test-history/test-history.component.html

```diff
@@ -33,7 +33,7 @@
           </span>
         </td>
         <td id="{{ 'test-expiryDate-tRes-' + i + '-' + ii }}" class="govuk-table__cell">
-          <span *ngIf="testType.testExpiryDate; else hyphen">
+          <span *ngIf="testType.testExpiryDate && testType.testExpiryDate.trim()!==''; else hyphen">
```

Comment:
just this `testType.testExpiryDate`  should be enough i think

Reply:
This is a bug fix because backend is sending " " in some cases.